source/plugins/ruby/KubernetesApiClient.rb

```diff
@@ -221,6 +224,16 @@ def getClusterId(env = ENV)
       return @@ClusterId
     end
 
+    def getUserAgent()
+      return @@userAgent if !@@userAgent.nil?
+      begin
+          @@userAgent = "ama-logs/#{ENV['AGENT_VERSION'].nil? ? '0.0.0' : ENV['AGENT_VERSION']} (#{ENV['OS_TYPE'].nil? ? 'linux' : ENV['OS_TYPE']}; Ruby #{RUBY_PLATFORM})"
+      rescue => error
+        @Log.warn("KubernetesAPIClient::getUserAgent : getUserAgent failed: #{error}")
+      end
+      return "ama-logs/0.0.0 ()"
```

Comment:
actually, there is bug. The return value should be @@userAgent otherwise very first case, this will be "ama-logs/0.0.0 ()"